design/assets.unity.md

```diff
@@ -0,0 +1,157 @@
+Unity Asset Management
+========================
+
+Goals:
+
+1. Minimize asset downloading and conversion/preprocessing.
+2. Balance memory footprint against load performance, and ensure no unreferenced assets stay in memory.
+2. Maximize asset reuse, so engines can instance and optimize draw calls.
+
+
+IAssetCache
+------------
+
+This is a system-wide class instance that is responsible for caching assets beyond the lifetime of a single MRE
+instance. This could possibly be backed by persistent storage instead of memory. This is primarily intended
+for assets loaded via an HTTP request.
+
+### StoreAssets
+
+```cs
+void StoreAssets(
+	string uri,
+	IEnumerable<UnityEngine.Object> assets = null,
+	string version = null
+);
+```
+
+If all arguments are not null, and either the cache contains no assets for the resource, or the cached version
+does not match what is provided, this method stores all provided assets in the cache, overwriting any currently
+cached assets. Otherwise, the assets and version arguments are ignored, and the internal reference counter for this
+resource is decremented.
+
+* `uri` - The resource identifier
+* `assets` - The collection of assets generated from the given resource
+* `version` - The version of the loaded resource. Will typically be the HTTP response's `ETag` header.
+
+### LeaseAssets
+
+```cs
+Task<IEnumerable<UnityEngine.Object>> LeaseAssets(
+	string uri,
+	string ifMatches = null
```

Comment:
Is this a specific version with a fixed format or just some generic filter? Please update the wording to reflect, what it actually is. If it's a version, then parameter named 'version' would be clearer.

Reply:
You mean update the parameter name? It's already described in the section below.

Reply:
How would the MRE runtime know when it is appropriate to provide a value for `ifMatches`?

Reply:
To Olga's point, wouldn't `matchesVersion` be a more descriptive parameter name as well?

Reply:
@eanders-MS There are two workflows this API supports:
1. Get version from cache, make a `GET` request with the `If-Matches` header, and pull from cache if 304 is returned.
2. Get version from server via `HEAD` request, pull from cache with `ifMatches`, then make a `GET` request if null is returned.

Right now I'm planning on using flow 1, and this argument is for flow 2 as a hedge against future use cases.

@olga-microsoft @tombuMS I named the argument this to make the comparison to the `If-Matches` HTTP header, but if it's confusing I'm happy to rename it.

Reply:
Does this parameter follow If-Match specification, can it have multiple values or be asterisk?
I suggest ifMatchesVersion, since you are using 'version' later in the spec to clarify that you are talking about the same string.

Reply:
That seems reasonable.